python/nano/src/bigdl/nano/pytorch/onnx/onnxrt_inference.py

```diff
@@ -28,9 +28,147 @@
                             '_build_ortsess',
                             'update_ortsess',
                             'predict_step',
+                            'on_fit_start',
                             'inference']
 
 
+# internal function to build an ortsess
+def _build_ortsess(self,
+                    input_sample=None,
+                    file_path="model.onnx",
+                    sess_options=None,
+                    **kwargs):
+    '''
+    Internal function to build a ortsess and bind to the lightningmodule.
+
+    :param input_sample: torch.Tensor for the model tracing.
+    :param file_path: The path to save onnx model file.
+    :param sess_options: ortsess options in ort.SessionOptions type
+    :param **kwargs: will be passed to torch.onnx.export function.
+    '''
+
+    if input_sample is None and self.example_input_array is not None:
+        input_sample = self.example_input_array  # use internal example_input_array
+    else:
+        self.example_input_array = input_sample  # set example_input_array for future usage
+
+    assert input_sample is not None,\
+        'You should set either input_sample or self.example_input_array'
+
+    default_onnx_export_args = {'export_params': True,
+                                'opset_version': 10,
+                                'do_constant_folding': True,
+                                'input_names': ['input'],
+                                'output_names': ['output'],
+                                'dynamic_axes': {'input': {0: 'batch_size'},
+                                                    'output': {0: 'batch_size'}}}
+    default_onnx_export_args.update(kwargs)
+
+    self.to_onnx(file_path,
+                 input_sample,
+                 **default_onnx_export_args)
+
+    self._ortsess = ort.InferenceSession(file_path, sess_options=sess_options)
+    self._ortsess_up_to_date = True
+
+
+# external method to update(& rebuild) ortsess
+def update_ortsess(self,
+                    input_sample=None,
+                    file_path="model.onnx",
+                    sess_options=None,
+                    **kwargs):
+    '''
+    Update the onnxruntime session options and rebuild the session.
+    Users may also want to call this method before `inference(..., onnx=True`)`
+    to avoid implicit building.
+
+    :param input_sample: torch.Tensor for the model tracing.
+    :param file_path: The path to save onnx model file.
+    :param sess_options: ortsess options in ort.SessionOptions type.
+    :param **kwargs: will be passed to torch.onnx.export function.
+    '''
+    self._build_ortsess(input_sample=input_sample,
+                        file_path=file_path,
+                        sess_options=sess_options,
+                        **kwargs)
+
+
+# inference (new API to unifying users' inference method)
+def inference(self,
+                input_data,
+                batch_size=None,
+                file_path="model.onnx",
```

Comment:
what if a user uses another file_path if ortsess is up to date?

Reply:
good catch! Now, nothing will happen since ortsess will not be built again.

I may remove this param since users may not rely on this inference method to export their onnx file. They always have `.to_onnx()`, the built-in method in pytorch lightning.